pkg/githubclient/client.go

```diff
@@ -0,0 +1,264 @@
+package githubclient
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/configuration-tools-for-gitops/pkg/log"
+	"github.com/google/go-github/v51/github"
+	"golang.org/x/oauth2"
+)
+
+type Github struct {
+	client              *github.Client
+	ctx                 context.Context
+	owner               string
+	repo                string
+	base                string
+	head                string
+	reconcileBranchName string
+}
+
+func New(token, owner, repo, base, head, reconcileBranchName string, ctx context.Context) (*Github, error) {
+	//Authenticate with Github
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
+	return &Github{
+		client,
+		ctx,
+		owner,
+		repo,
+		base,
+		head,
+		reconcileBranchName,
+	}, nil
+}
+
+func (gh *Github) Merge(dryRun bool) error {
+	err := gh.mergeBranches()
+	if err != nil {
+		if strings.Contains(err.Error(), "Merge conflict") {
+			return gh.handleMergeConflict(dryRun)
+		} else {
+			return fmt.Errorf("failed to merge branches: %v", err)
+		}
+	} else {
+		if dryRun {
+			log.Sugar.Debug("No merge conflicts found")
+			return nil
+		}
+		log.Sugar.Info("Merged successfully")
```

Comment:
please don't use log.Sugar directly in general packages.
This creates a tight dependency on the `log` package that we are using.
If you want to use the log functionality in the package, then pass a Logger interface to the constructor and use it in the package

Reply:
for an example see the exec package [here](https://github.wdf.sap.corp/ICN-ML/aicore/blob/2565d11c7bfdbab9298ae18c850131568e1da6c8/system-services/platform/pkg/exec/exec.go#L55)